iml-gui/crate/src/components/action_dropdown.rs

```diff
@@ -0,0 +1,273 @@
+                ],
+            ]),
+        ]),
+    )
+    .map_msg(move |m| IdMsg(id, Msg::ConfirmModal(m)))
+}
+
+pub fn view(id: u32, model: &Model) -> Node<IdMsg> {
+    let cls = class![
+        C.bg_blue_500,
+        C.hover__bg_blue_700,
+        C.text_white,
+        C.font_bold,
+        C.py_2,
+        C.px_4,
+        C.rounded,
+    ];
+
+    let disabled_cls = class![C.opacity_50, C.cursor_not_allowed];
+
+    match &model.state {
+        State::Activating => button![
+            cls.merge_attrs(disabled_cls),
+            "Actions",
+            font_awesome(class![C.w_4, C.h_4, C.inline, C.ml_1, C.pulse], "spinner"),
+        ],
+        State::Inactive => button![
+            cls,
+            "Actions",
+            font_awesome(class![C.w_4, C.h_4, C.inline, C.ml_1], "chevron-down"),
+            simple_ev(Ev::MouseMove, IdMsg(id, Msg::StartFetch))
+        ],
+        State::Active => {
+            if model.actions.is_empty() {
+                button![cls.merge_attrs(disabled_cls), "No Actions"]
+            } else {
+                span![
+                    class![C.relative, C.inline_block],
+                    button![
+                        cls,
+                        "Actions",
+                        font_awesome(class![C.w_4, C.h_4, C.inline, C.ml_1], "chevron-down"),
+                        simple_ev(Ev::Click, IdMsg(id, Msg::WatchChange))
+                    ],
+                    dropdown::wrapper_view(
+                        class![C.z_30, C.w_64],
+                        Placement::Bottom,
+                        model.watching.is_open(),
+                        model
+                            .actions
+                            .iter()
+                            .map(|x| {
+                                let x2 = Arc::clone(x);
+                                div![
+                                    tooltip::container(),
+                                    dropdown::item_view(a![x.verb,]),
+                                    tooltip::view(&x.long_description, Placement::Left),
+                                    mouse_ev(Ev::Click, move |_| IdMsg(id, Msg::ActionSelected(x2)))
+                                ]
+                            })
+                            .collect::<Vec<_>>()
+                    )
+                ]
+            }
+        }
+        State::Confirming(title, body) => div![
+            button![
+                cls.merge_attrs(disabled_cls),
+                "Actions",
+                font_awesome(class![C.w_4, C.h_4, C.inline, C.ml_1, C.pulse], "spinner"),
+            ],
+            confirm_modal_view(id, title, body),
+        ],
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_composite_ids_to_query_string() {
+        let query_string = composite_ids_to_query_string(&[CompositeId(57, 3), CompositeId(49, 1)]);
+
+        assert_eq!(query_string, "composite_ids=49:1&composite_ids=57:3");
```

Comment:
I see sorting was removed, which is good, and now it looks the order will be `composite_ids=57:3&composite_ids=49:1`